tests/unittests/tests-phydat/tests-phydat.c

```diff
@@ -14,46 +15,103 @@
 #define ENABLE_DEBUG (0)
 #include "debug.h"
 
+/* Default is smaller implementation */
+#ifndef PHYDAT_FIT_TRADE_PRECISION_FOR_ROM
+#define PHYDAT_FIT_TRADE_PRECISION_FOR_ROM 1
+#endif
+
 static void test_phydat_fit(void)
 {
-    /* verify that these big numbers are scaled to fit in phydat_t::val which is int16_t */
-    long val0 =   100445;
-    long val1 =  2000954;
-    long val2 = 30000455;
-    long val4 =  1234567;
-    phydat_t dat;
-    dat.scale = -6;
-    dat.unit = UNIT_V;
-    uint8_t res = phydat_fit(&dat, val0, 0, 0);
-    /* Check that the result was rescaled to 10044e-5 */
-    /* The scaled number is rounded toward zero */
-    TEST_ASSERT_EQUAL_INT(1, res);
-    TEST_ASSERT_EQUAL_INT(UNIT_V, dat.unit);
-    TEST_ASSERT_EQUAL_INT(-5, dat.scale);
-    TEST_ASSERT_EQUAL_INT( 10044, dat.val[0]);
-    /* Fit the next value in the phydat vector */
-    res = phydat_fit(&dat, val1, 1, res);
-    TEST_ASSERT_EQUAL_INT(2, res);
-    TEST_ASSERT_EQUAL_INT(UNIT_V, dat.unit);
-    TEST_ASSERT_EQUAL_INT(-4, dat.scale);
-    TEST_ASSERT_EQUAL_INT(  1004, dat.val[0]);
-    TEST_ASSERT_EQUAL_INT( 20009, dat.val[1]);
-    /* Fit the third value in the phydat vector */
-    res = phydat_fit(&dat, val2, 2, res);
-    TEST_ASSERT_EQUAL_INT(3, res);
-    TEST_ASSERT_EQUAL_INT(UNIT_V, dat.unit);
-    TEST_ASSERT_EQUAL_INT(-3, dat.scale);
-    TEST_ASSERT_EQUAL_INT(   100, dat.val[0]);
-    TEST_ASSERT_EQUAL_INT(  2000, dat.val[1]);
-    TEST_ASSERT_EQUAL_INT( 30000, dat.val[2]);
-    /* Overwrite the second value in the phydat vector */
-    res = phydat_fit(&dat, val4, 1, res);
-    TEST_ASSERT_EQUAL_INT(3, res);
-    TEST_ASSERT_EQUAL_INT(UNIT_V, dat.unit);
-    TEST_ASSERT_EQUAL_INT(-3, dat.scale);
-    TEST_ASSERT_EQUAL_INT(   100, dat.val[0]);
-    TEST_ASSERT_EQUAL_INT(  1234, dat.val[1]);
-    TEST_ASSERT_EQUAL_INT( 30000, dat.val[2]);
+    /* Input values for each test: */
+    static const int32_t values[][3] = {
+        { 100445, -1, -1 },
+        { -5, 2000954, 3 },
+        { 30000449, -30000450, 30000500 },
+        { -30000449, -30000499, -30000500 },
+        { 0, 0, 1234567 },
+        { 32768, 32768, 32768 },
+        { 32767, 32767, 32767 },
+        { 32766, 32766, 32766 },
+        { -32769, -32769, -32769 },
+        { -32768, -32768, -32768 },
+        { -32767, -32767, -32767 },
+        { -32766, -32766, -32766 },
```

Comment:
This should not be required. The [C99 standard](http://www.open-std.org/jtc1/sc22/wg14/www/docs/n1256.pdf) defines in 6.4.4.1 list item 5 states that for decimal numbers the smallest type of `int`, `long`, `long long` should be used that the number fits into. An example where the postfix is relevant (assuming `sizeof(int) == 2`, `sizeof(long) == 4`) would be:

``` C
long foo = 42<<16; //< result would be 0
long bar = 42L<<16; //< result would be 2752512
```

Here the 42 would have the type `int` and shifting it by 16 would result in `0`, as all bis would be shifted out of the 16 bits of the `int` type. Only after the shifting is performed, it would be converted to `long`. The assignment in `bar` would work, as here the type `long` is forced by the postfix.

Of course adding the `L`-postfix would not hurt. Should I add it anyway?

Reply:
Thanks for the explanation. My comment is resolved, no change needed.